### PR TITLE
Connect API

### PR DESCRIPTION
**Pull request by Laylo309**
- [x] Refactor your add book and remove book features to persist changes in the server.
- [x] Refactor the redux code of your application using middleware to make async requests to the API.
- [x] _Add book_ and _remove book_ work in the same way after the refactor.

### CODE REVIEW

**Review by mke2111 — CHANGES_REQUESTED** [review]
### Hey @Laylo309, 

**This is a great project you have going on, well-done lad**. 

   
**However, there are a few sections to improve before your project is approved.** 🔴 
<hr>

## Required Changes ♻️
_Check the inline comments under the review for sections to improve._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/Books.js [hunk 1]
```diff
@@ -1,47 +1,49 @@
...
+  const booksList = booksData.map((book) => (
+    <BooksContainer
+      title={book.title}
+      author={book.author}
+      category={book.category}
+      key={booksData.indexOf(book)}
+      removeBttn={() => {
+        deleteBook(book);
+      }}
+    />
```
Comment (mke2111) [inline src/components/Books.js:41]: - [ ] When I create a new book entry, I get the following error.
Screenshot from 2021-10-02 04-47-35

Ensure that you are using the correct name as imported.

Comment (Laylo309) [reply src/components/Books.js:41]: Fixed ✅

**Review by vichuge — APPROVED** [review]
# Approved project! ✅ 🥳 🎈 🎉🎉🎉🎉🎉🎉🎉🎉 

Hi there, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

InbornCandidHornet-size_restricted

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**